src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/profile/items/museum/MuseumAPI.kt

```diff
@@ -53,6 +61,33 @@ object MuseumAPI {
     fun getItemsOnCategory(category: MuseumCategory): List<ItemStack> = MuseumStorage.getItemsOnCategory(category)
     fun getItemsWithCategory(): Map<MuseumCategory, List<ItemStack>> = MuseumStorage.getItemsWithCategory()
 
+    /** Returns `true` if the item can be donated to museum. Ignores special items. */
+    fun isMuseumItem(id: SkyBlockId): Boolean = MuseumData.isMuseumItem(id)
+
+    /** Returns `true` if the item has been donated to museum. Ignores special items. */
+    fun isDonated(id: SkyBlockId): Boolean {
+        val skyblockId = id.skyblockId
+        val data = ItemData.getItemData(skyblockId)?.museumData ?: return false
+        return if (data.category == MuseumCategory.ARMOR_SETS) {
+            val armorSets = data.armorSets
+            armorSets.any { MuseumStorage.hasDonatedArmorSet(it) } ||
+                data.parents.values.any { parent ->
+                    parent !in armorSets &&
+                        (MuseumStorage.hasDonatedArmorSet(parent) ||
+                            MuseumData.getArmorSetFromId(parent)?.firstOrNull()?.let { isDonated(SkyBlockId.item(it)) } == true)
+                }
+        } else MuseumStorage.hasDonatedItem(skyblockId) || data.parents.values.any { isDonated(SkyBlockId.item(it)) }
```

Comment:
```suggestion
        if (data.category == MuseumCategory.ARMOR_SETS) {
            val armorSets = data.armorSets
            val donated = armorSets.any(MuseumStorage::hasDonatedArmorSet)
            val parentsDonated = data.parents.values.any { parent ->
                    if (parent in armorSets) return@any false
                    if (MuseumStorage.hasDonatedArmorSet(parent)) return@any true 
                    val firstParentArmorPiece = MuseumData.getArmorSetFromId(parent)?.firstOrNull() ?: return@any false
                    return@any isDonated(SkyBlockId.item(firstParentArmorPiece))
            }
            return donated || parentsDonated
        } 
        
        return MuseumStorage.hasDonatedItem(skyblockId) || data.parents.values.any { isDonated(SkyBlockId.item(it)) }
```

wrote this all in github so this should be reviewed in ide and tested